stdlib/experimental/dynamic/dynamic.go

```diff
@@ -21,10 +27,94 @@ var dynamicConv = values.NewFunction(
 			return nil, err
 		}
 
+		// Nothing to do if the incoming value is already a dynamic
+		if v.Type().Nature() == semantic.Dynamic {
+			v.Retain()
+			return v, nil
+		}
+
 		// FIXME(onelson): do we need to special case nulls here?
 		//   Ex: if the input is null, should the output be null or a
 		//   Dynamic with a null in it? My sense is the latter.
 		return values.NewDynamic(v), nil
 	},
 	false,
 )
+
+var asArray = values.NewFunction(
+	"asArray",
+	runtime.MustLookupBuiltinType("experimental/dynamic", "asArray"),
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		arguments := interpreter.NewArguments(args)
+		v, err := arguments.GetRequired("v")
+		if err != nil {
+			return nil, err
+		}
+
+		if v.IsNull() {
+			panic("TODO")
```

Comment:
Should this just be an error as below? Or propagate the `null`?

Reply:
Well color me embarrassed. Let me clean this up :cold_sweat: Yeah, it should be something like that.

Reply:
Starting out with an error following the "error if the input is not an array" policy. 
I'm not sure our static checks save us from untyped nulls getting through to a receiver like this, so I guess the check is still necessary.